File: Java/Leetcode/01_Array/27_RemoveElement.java
package Leetcode;

class RemoveElement {

    public int removeElement(int[] nums, int val) {

        int slow = 0, fast = 0;
        int len = nums.length;

        while (fast < len) {

            if (nums[fast] == val)
                fast++;
            else
                nums[slow++] = nums[fast++];

        }
        
        return slow;

    }

}

File: C&C++/Leetcode/27_RemoveElement.c
int removeElement(int *nums, int numsSize, int val)
{

    int i = -1, j = -1;
    while (++j < numsSize)
    {
        if (nums[j] != val)
            nums[++i] = nums[j];
    }

    return i + 1;
}